internal/lint/pod.go

```diff
@@ -117,10 +117,10 @@ func (s *Pod) checkNPs(ctx context.Context, pod *v1.Pod) {
 			continue
 		}
 		if labelsMatch(&np.Spec.PodSelector, pod.Labels) {
-			if s.checkIngresses(ctx, pod, np.Spec.Ingress) {
```

Comment:
The pod has been selected by the `PodSelector`, therefore the ingress & egress (if present) apply to it.  We don't need the Ingress & Egress rules themselves to target the pod.